kernel/common/parse_ip.rs

```diff
@@ -1,37 +1,36 @@
 
 /// Get the port from a string (ip)
 pub fn parse_port(string: &str) -> &str {
-    let mut a = 0;
-    let mut b = 0;
+    let a = match string.find(':') {
+        Some(pos) => pos + 1,
+        None => 0,
+    };
 
-    for (n, c) in string.chars().enumerate() {
+    let mut b = 0;
+    for c in string.chars().skip(a) {
         match c {
             '0' | '1' |
             '2' | '3' |
             '4' | '5' |
             '6' | '7' |
             '8' | '9' => b += 1,
-            ':' => a = n + 1,
             _ => break,
         }
     }
 
     &string[a..b + 1]
-
 }
 
 /// Get the host from a string (ip)
 pub fn parse_host(string: &str) -> &str {
-    let mut b = 0;
-
     // TODO: Username/Password syntax
-    for (n, c) in string.chars().enumerate() {
-        match c {
-            ':' | '/' => break,
-            _ => b += 1,
-        }
-    }
-
-    &string[..b + 1]
-
+    let pos = match string.find(':') {
+        Some(pos) => pos + 1,
+        None => match string.find('/') {
+            Some(pos) => pos + 1,
```

Comment:
Sure, just keep in mind to account for the possibility that if it can't find those two chars, the slice should be `&string[..b]`, not `&string[..b + 1]`